src/hierarchical_interleaved_bloom_filter.cpp

```diff
@@ -219,10 +219,19 @@ hierarchical_interleaved_bloom_filter::hierarchical_interleaved_bloom_filter(con
                                     return count == 0u;
                                 }));
 
+    std::vector<size_t> positions = [&kmer_counts]()
+    {
+        std::vector<size_t> ps;
+        ps.resize(kmer_counts.size());
+        std::iota(ps.begin(), ps.end(), 0);
```

Comment:
If we don't do a function, we should definitely change this in all occurrences:

```suggestion
        std::iota(ps.begin(), ps.end(), size_t{});
```